src/sentry/issues/auto_source_code_config/stacktraces.py

```diff
@@ -1,41 +1,42 @@
+from __future__ import annotations
+
 import logging
-from collections.abc import Mapping, Sequence
-from typing import Any, TypedDict
+from typing import Any
 
 from sentry.db.models.fields.node import NodeData
 from sentry.utils.safe import get_path
 
-logger = logging.getLogger(__name__)
-
+from .constants import PROCESS_ALL_FRAMES
 
-class Stacktrace(TypedDict):
-    stacktrace: Mapping[str, Sequence[Mapping[str, Any]]]
+logger = logging.getLogger(__name__)
 
 
-def identify_stacktrace_paths(data: NodeData | Stacktrace) -> list[str]:
-    """
-    Get the stacktrace_paths from the event data.
-    """
-    stacktraces = get_stacktrace(data)
-    stacktrace_paths = set()
+def get_frames_to_process(
+    data: NodeData | dict[str, Any], platform: str | None = None
+) -> list[dict[str, Any]]:
+    """It flattens all processableframes from the event's data."""
+    stacktraces = get_stacktraces(data)
+    frames_to_process = []
     for stacktrace in stacktraces:
         frames = stacktrace["frames"]
         for frame in frames:
-            if frame is None:
-                continue
+            if platform in PROCESS_ALL_FRAMES:
+                frames_to_process.append(frame)
+
+            elif frame.get("in_app") and frame.get("filename"):
+                frames_to_process.append(frame)
 
-            if frame.get("in_app") and frame.get("filename"):
-                stacktrace_paths.add(frame["filename"])
-    return list(stacktrace_paths)
+    return list(frames_to_process)
 
 
-def get_stacktrace(data: NodeData | Stacktrace) -> list[Mapping[str, Any]]:
+def get_stacktraces(data: NodeData | dict[str, Any]) -> list[dict[str, Any]]:
     exceptions = get_path(data, "exception", "values", filter=True)
     if exceptions:
         return [e["stacktrace"] for e in exceptions if get_path(e, "stacktrace", "frames")]
 
     stacktrace = data.get("stacktrace")
     if stacktrace and stacktrace.get("frames"):
+        logger.warning("Investigate if we use this code path in production.")
```

Comment:
The only reason I believe we need this block is because of the way I wrote the tests. If this error doesn't happen in production I will remove it.

Reply:
@MichaelSun48 me as well! `logger.warning()` captures a message in Sentry with a level of warning 👍🏻